core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -352,9 +352,9 @@ public String getCarbonDataDirectoryPath(String partitionId, String segmentId) {
    * @return gets data file name only with out path
    */
   public String getCarbonDataFileName(Integer filePartNo, Integer taskNo, int bucketNumber,
-      String factUpdateTimeStamp) {
-    return DATA_PART_PREFIX + filePartNo + "-" + taskNo + "-" + bucketNumber + "-"
-        + factUpdateTimeStamp + CARBON_DATA_EXT;
+      int taskExtension, String factUpdateTimeStamp) {
+    return DATA_PART_PREFIX + filePartNo + "-" + taskNo + "_" + taskExtension + "-" + bucketNumber
```

Comment:
Should we give some prefix to the taskExtension to make it more clear? I feel it is too many number